broadinstitute/single_cell_portal_core
# MVP Legend search (SCP-4622)

Add a feature flagged legend search for plot legends that have more than 30 entries.

To test:
- Pull this branch
- Run the DB migration with: `bin/rails db:migrate`
- Boot up a localhost instance as usual
- log in and go to the feature flag section
- set the feature flag for legend search to true for your user
- go to a study that is visualizable that has an annotation with more than 30 entries (I looked at `Study: Aging mouse` using `cell_type_by_age` annotation)
- Confirm that legend search appears and you can filter down the legend results and the `hide all` and `show all` buttons continue to work as expected.

NOTE:
- Customizable colors UI is currently broken so I have disabled customizing colors when the legend search is enabled for now to not exacerbate the problem and not scope creep this ticket. There is a different ticket in sprint to fix the buggy custom colors.


https://user-images.githubusercontent.com/54322292/191302380-8ba4e4db-abee-4701-a205-0f5971f19dc3.mov
